gobreaker.go

```diff
@@ -18,6 +19,25 @@ const (
 	StateOpen
 )
 
+// ErrTooManyRequests is returned when the CB state is half open and the requests count is over the cb maxRequests
+var ErrTooManyRequests = errors.New("too many requests")
+
+// ErrCircuitOpen is returned when the CB state is open
+var ErrCircuitOpen = errors.New("circuit breaker is open")
+
+// ErrCircuitBreaker is a generic error returned from the CB, wrapping the error and containing the name
+type ErrCircuitBreaker struct {
```

Comment:
Style-wise, since this is a type, it should be named `CircuitBreakerError`. `ErrFoo` is for error variables, not types.

Reference: https://talks.golang.org/2014/names.slide#15.

You could also consider just `Error`, since the full name to users of this package would otherwise be `gobreaker.CircuitBreakerError` which stutters.

Reply:
You're right. Thank you! 👍